addons/account/models/account_account.py

```diff
@@ -460,25 +466,40 @@ def _compute_opening_debit_credit(self):
 
     @api.depends('code')
     def _compute_account_type(self):
-        """ Compute the account type based on the account code.
-        Search for the closest parent account code and sets the account type according to the parent.
-        If there is no parent (e.g. the account code is lower than any other existing account code),
-        the account type will be set to 'asset_current'.
+        """
+            Compute the account type based on the account code.
+            Search for the closest parent account code and sets the account type according to the parent.
+            If there is no parent (e.g. the account code is lower than any other existing account code),
+            the account type will be set to 'asset_current'.
         """
         accounts_to_process = self.filtered(lambda r: r.code and not r.account_type)
+        self._helper_closest_parent_account(accounts_to_process, 'account_type', 'asset_current')
+
+    @api.depends('code')
+    def _compute_account_tags(self):
+        """
+            Compute the account tags based on the account code.
+            Search for the closest parent account code and sets the account tags according to the parent.
+            If there is no parent (e.g. the account code is lower than any other existing account code),
+            the account tags will be an empty list.
+        """
+        accounts_to_process = self.filtered(lambda r: r.code and not r.tag_ids)
+        self._helper_closest_parent_account(accounts_to_process, 'tag_ids', [])
+
+    def _helper_closest_parent_account(self, accounts_to_process, field_name, default_value):
         all_accounts = self.search_read(
             domain=[('company_id', 'in', accounts_to_process.company_id.ids)],
-            fields=['code', 'account_type', 'company_id'],
+            fields=['code', field_name, 'company_id'],
             order='code',
         )
```

Comment:
I know it's not your code directly but shouldn't we take inactive account as well?